modules/channel-teams/src/backend/client.ts

```diff
@@ -141,20 +141,28 @@ If you have a restricted app, you may need to specify the tenantId also.`
   }
 
   private _sendIncomingEvent = async (activity: Activity, threadId: string) => {
-    const { text, from, type } = activity
-
-    await this.bp.events.sendEvent(
-      this.bp.IO.Event({
-        botId: this.botId,
-        channel: 'teams',
-        direction: 'incoming',
-        payload: { text },
-        preview: text,
-        threadId,
-        target: from.id,
-        type
-      })
-    )
+    const {
+      text,
+      from: { id: userId },
+      type
+    } = activity
+
+    let convoId = await this.bp.experimental.conversations
+      .forBot(this.botId)
+      .getLocalId('teams', `${threadId}&${userId}`)
```

Comment:
`&` should work as a seperator. It's worth noting that teams ids are really long. A conversation and user id combined make 220 characters which is dangerously close to the limit of 256 for mapping. Maybe the mapping table should extend the length of its foreign column to 512 just to be sure

Reply:
I thought that they were only using GUIDs which, if I'm correct, only contain hyphens. Am I missing something here?